JAICore/jaicore-basic/src/main/java/jaicore/basic/kvstore/KVStore.java

```diff
@@ -266,26 +332,31 @@ public boolean matches(final Map<String, String> selection) {
 		boolean doesNotMatchAllSelectionCriteria = selection.entrySet().stream().anyMatch(x -> {
 			boolean isEqual = this.getAsString(x.getKey()).equals(x.getValue());
 
+			if (!x.getValue().contains("*")) {
+				return !x.getValue().equals(this.getAsString(x.getKey()));
+			}
+
 			String[] exprSplit = x.getValue().split("\\*");
 			String currentValue = x.getValue();
 			boolean matchesPattern = true;
 
 			for (int i = 0; i < exprSplit.length; i++) {
 				if (i == 0 && !x.getValue().startsWith("*") && !currentValue.startsWith(exprSplit[i])) {
 					matchesPattern = false;
-					break;
 				}
 
 				if (currentValue.contains(exprSplit[i])) {
-					currentValue.replaceFirst("(" + exprSplit[i] + ")", "#$#");
+					currentValue = currentValue.replaceFirst("(" + exprSplit[i] + ")", "#$#");
 					currentValue = currentValue.split("#$#")[1];
 				} else {
 					matchesPattern = false;
-					break;
 				}
 
 				if (i == (exprSplit.length - 1) && !x.getValue().endsWith("*") && !currentValue.endsWith(exprSplit[i])) {
 					matchesPattern = false;
+				}
+

```

Comment:
I know that this is SonarQube convention. But is this really more readable than with the breaks in place? We should discuss whether we want to keep this rule.